Source/Scene/Cesium3DTile.js

```diff
@@ -0,0 +1,1018 @@
+            this._debugViewerRequestVolume = this._debugViewerRequestVolume && this._debugViewerRequestVolume.destroy();
+        }
+    };
+
+    function applyDebugSettings(tile, tileset, frameState) {
+        var hasContentBoundingVolume = defined(tile._header.content) && defined(tile._header.content.boundingVolume);
+
+        var showVolume = tileset.debugShowBoundingVolume || (tileset.debugShowContentBoundingVolume && !hasContentBoundingVolume);
+        if (showVolume) {
+            if (!defined(tile._debugBoundingVolume)) {
+                var color = tile._finalResolution ? (hasContentBoundingVolume ? Color.WHITE : Color.RED) : Color.YELLOW;
+                tile._debugBoundingVolume = tile._boundingVolume.createDebugVolume(color);
+            }
+            tile._debugBoundingVolume.update(frameState);
+        } else if (!showVolume && defined(tile._debugBoundingVolume)) {
+            tile._debugBoundingVolume = tile._debugBoundingVolume.destroy();
+        }
+
+        if (tileset.debugShowContentBoundingVolume && hasContentBoundingVolume) {
+            if (!defined(tile._debugContentBoundingVolume)) {
+                tile._debugContentBoundingVolume = tile._contentBoundingVolume.createDebugVolume(Color.BLUE);
+            }
+            tile._debugContentBoundingVolume.update(frameState);
+        } else if (!tileset.debugShowContentBoundingVolume && defined(tile._debugContentBoundingVolume)) {
+            tile._debugContentBoundingVolume = tile._debugContentBoundingVolume.destroy();
+        }
+
+        if (tileset.debugShowViewerRequestVolume && defined(tile._viewerRequestVolume)) {
+            if (!defined(tile._debugViewerRequestVolume)) {
+                tile._debugViewerRequestVolume = tile._viewerRequestVolume.createDebugVolume(Color.YELLOW);
+            }
+            tile._debugViewerRequestVolume.update(frameState);
+        } else if (!tileset.debugShowViewerRequestVolume && defined(tile._debugViewerRequestVolume)) {
+            tile._debugViewerRequestVolume = tile._debugViewerRequestVolume.destroy();
+        }
+
+        var debugColorizeTilesOn = tileset.debugColorizeTiles && !tile._debugColorizeTiles;
+        var debugColorizeTilesOff = !tileset.debugColorizeTiles && tile._debugColorizeTiles;
+
+        if (debugColorizeTilesOn) {
+            tile._debugColorizeTiles = true;
+            tile.color = tile._debugColor;
+        } else if (debugColorizeTilesOff) {
+            tile._debugColorizeTiles = false;
+            tile.color = Color.WHITE;
+        }
+
+        if (tile._colorDirty) {
+            tile._colorDirty = false;
+            tile._content.applyDebugSettings(true, tile._color);
+        }
+
+        if (debugColorizeTilesOff) {
+            tileset.makeStyleDirty(); // Re-apply style now that colorize is switched off
+        }
+    }
+
+    function updateContent(tile, tileset, frameState) {
```

Comment:
Some high level description for some of these private functions could go a long way to helping new maintainers understand the code.